src/index.ts

```diff
@@ -90,14 +90,14 @@ class Resolver {
    * @param {string} hostname - host name to resolve
    * @param {string} [rrType = 'A'] - resource record type
    */
-  async resolve (hostname: string, rrType = 'A') {
+  async resolve (hostname: string, rrType = 'A'): Promise<string[] | string[][]> {
```

Comment:
Instead of returning `Promise<string[] | string[][]>` we can use method overloading to return a specific type based on the `rrType` parameter.  This way the user doesn't have to work out what they were given:

```suggestion
  async resolve (hostname: string, rrType: 'TXT'): Promise<string[][]>
  async resolve (hostname: string, rrType: 'A' | 'AAAA'): Promise<string[]>
  async resolve (hostname: string): Promise<string[]>
  async resolve (hostname: string, rrType: string = 'A'): Promise<string[] | string[][]> {
```